ArduCopter/commands_logic.cpp

```diff
@@ -1084,13 +1084,13 @@ bool Copter::ModeAuto::verify_within_distance()
 // verify_yaw - return true if we have reached the desired heading
 bool Copter::ModeAuto::verify_yaw()
 {
-    // set yaw mode if it has been changed (the waypoint controller often retakes control of yaw as it executes a new waypoint command)
-    if (auto_yaw_mode != AUTO_YAW_LOOK_AT_HEADING) {
-        set_auto_yaw_mode(AUTO_YAW_LOOK_AT_HEADING);
-    }
+    // set yaw mode if it has been changed (the waypoint controller
+    // often retakes control of yaw as it executes a new waypoint
+    // command)
+    auto_yaw.set_mode(AUTO_YAW_FIXED);
 
     // check if we are within 2 degrees of the target heading
-    if (labs(wrap_180_cd(ahrs.yaw_sensor-copter.yaw_look_at_heading)) <= 200) {
```

Comment:
On my SITL simulations, my YAW error is always around 2 deg. Maybe we should make this 200 a parameter, so that users that require more precision can change it ?